src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,222 @@
+        // Physics Reference 10.6)
+        BernoulliDistribution choose_f1g1(
+            (ipow<2>(0.5 - epsilon_min) - epsilon_min
+             + 0.25 * this->screening_phi1_aux(delta_min))
+            / (1.5 * this->screening_phi2_aux(delta_min)));
+
+        // Temporary sample values used in rejection
+        real_type reject_threshold;
+        do
+        {
+            if (choose_f1g1(rng))
+            {
+                // Used to sample from f1
+                epsilon = 0.5
+                          - (0.5 - epsilon_min)
+                                * std::cbrt(generate_canonical(rng));
+                CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+                // Calculate delta from element atomic number and sampled
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g1 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi1_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi1_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold > 0.0 && reject_threshold <= 1.0);
+            }
+            else
+            {
+                // Used to sample from f2
+                epsilon = epsilon_min
+                          + (0.5 - epsilon_min) * generate_canonical(rng);
+                CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+                // Calculate delta given the element atomic number and sampled
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g2 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi2_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi2_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold > 0.0 && reject_threshold <= 1.0);
+            }
+        } while (BernoulliDistribution(1.0 - reject_threshold)(rng));
+    }
+
+    // Construct interaction for change to primary (incident) particle (gamma)
+    Interaction result = Interaction::from_absorption();
+    result.secondaries = {secondaries, 2};
+
+    // Outgoing secondaries are electron and positron
+    secondaries[0].def_id = shared_.electron_id;
+    secondaries[1].def_id = shared_.positron_id;
+    secondaries[0].energy
+        = units::MevEnergy{(1.0 - epsilon) * inc_energy_.value()};
+    secondaries[1].energy = units::MevEnergy{epsilon * inc_energy_.value()};
+    // Select charges for child particles (e-, e+) randomly
+    if (BernoulliDistribution(0.5)(rng))
+    {
+        swap2<units::MevEnergy>(secondaries[0].energy, secondaries[1].energy);
```

Comment:
At some point I added a specialization for swapping `Quantity` objects, so you should be able to just replace this with 
```suggestion
        swap(secondaries[0].energy, secondaries[1].energy);
```